tka/key.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tka
+
+import (
+	"crypto/ed25519"
+	"errors"
+	"fmt"
+
+	"github.com/hdevalence/ed25519consensus"
+)
+
+// KeyKind describes the different varieties of a Key.
+type KeyKind uint8
+
+// Valid KeyKind values.
+const (
+	KeyInvalid KeyKind = iota
+	Key25519
+)
+
+func (k KeyKind) String() string {
+	switch k {
+	case KeyInvalid:
+		return "invalid"
+	case Key25519:
+		return "25519"
+	default:
+		return fmt.Sprintf("Key?<%d>", int(k))
+	}
+}
+
+// Key describes the public components of a key known to network-lock.
+type Key struct {
+	Kind KeyKind `cbor:"1,keyasint"`
+
+	// Votes describes the weight applied to signatures using this key.
+	// Weighting is used to deterministically resolve branches in the AUM
+	// chain (i.e. forks, where two AUMs exist with the same parent).
+	Votes uint `cbor:"2,keyasint"`
+
+	// Public encodes the public key of the key. For 25519 keys,
+	// this is simply the point on the curve representing the public
+	// key.
+	Public []byte `cbor:"3,keyasint"`
+
+	Meta map[string]string `cbor:"12,keyasint,omitempty"`
```

Comment:
This could really use a comment. What are the keys / values?

Reply:
Done. This is an arbitrary K:V container for human-friendly metadata about the key thats irrelevant to the operation of TKA. Name, Created, whatever.